api/all/src/main/java/io/opentelemetry/api/trace/ImmutableSpanContext.java

```diff
@@ -14,21 +14,30 @@
 abstract class ImmutableSpanContext implements SpanContext {
 
   static final SpanContext INVALID =
-      create(
+      createInternal(
           TraceId.getInvalid(),
           SpanId.getInvalid(),
           TraceFlags.getDefault(),
           TraceState.getDefault(),
           /* remote= */ false);
 
+  private static AutoValue_ImmutableSpanContext createInternal(
+      String traceId, String spanId, TraceFlags traceFlags, TraceState traceState, boolean remote) {
+    return new AutoValue_ImmutableSpanContext(
+        traceId, spanId, traceFlags, traceState, /* remote$= */ remote);
```

Comment:
Now that you calculate the isValid it can be a property in the AutoValue so no need to recalculate that or to use memoization.

Reply:
excellent idea. let's create an issue to track that optimization.

Reply:
#2752